ili2c-core/src/main/antlr/ch/interlis/ili2c/parser/interlis24.g

```diff
@@ -5066,18 +5091,58 @@ protected factor[Container ns,Container functionNs]
 			{ inspFactor.setRestriction(inspRestriction); 
 			}
 		)?) 
-	|	({
-			if(!(ns instanceof Viewable)){
-				reportError (formatMessage ("err_Container_currentIsNotViewable",
-				ns.toString()), LT(1).getLine());
-			}
+	| {ns instanceof Viewable}? ev = objectOrAttributePath[(Viewable)ns,functionNs]
+	{
+		if(ev == null){
+			reportError (formatMessage ("err_Container_currentIsNotViewable",
+			ns.toString()), LT(1).getLine());
 		}
-		ev=objectOrAttributePath[(Viewable)ns,functionNs]
-		)
+	}
+	| {ns instanceof Domain && ((Domain)ns).getType() instanceof ClassType}? ev = classTypeObjectOrAttributePath[(ClassType)(((Domain)ns).getType()),functionNs]
+	| {ns instanceof Domain}? ev = valueRefThis[(Domain)ns]
 	|	ev=constant[ns]
-		
 	;
 
+
+protected classTypeObjectOrAttributePath[ClassType classType, Container functionNs]
+	returns [ClassTypeObjectPath classTypeObjectPath]
+	{
+		classTypeObjectPath = null;
+		ObjectPath objectPath = null;
+		int start = mark();
+		Iterator<Viewable<?>> restrictions = classType.iteratorRestrictedTo();
+		List<ObjectPath> objectPaths = new ArrayList<ObjectPath>();
+	}
+	:
+	{!restrictions.hasNext()}?
+		objectPath = objectOrAttributePath[classType.isStructure() ? modelInterlis.ANYSTRUCTURE : modelInterlis.ANYCLASS, functionNs]
+		{
+			classTypeObjectPath = new ClassTypeObjectPath(new ObjectPath[] { objectPath });
+		}
+	|
+		objectPath = objectOrAttributePath[restrictions.next(), functionNs]
+		{
+			objectPaths.add(objectPath);
+			while (restrictions.hasNext()) {
+				rewind(start);
+				objectPath = objectOrAttributePath(restrictions.next(), functionNs);
+				objectPaths.add(objectPath);
+			}
+			classTypeObjectPath = new ClassTypeObjectPath(objectPaths.toArray(new ObjectPath[objectPaths.size()]));
+		}
```

Comment:
REQ: Prüfen, inwiefern bei STRUCTURE bzw. CLASS RESTRICTIONS das Parsen von Attribut Pfaden überhaupt sinnvoll ist.
THIS könnte ev. einfach zu 'STRUCTURE' evaluieren ähnlich wie bei `FUNCTION myClass (Object: ANYSTRUCTURE): STRUCTURE;
`